go/ql/lib/semmle/go/security/TaintedPathCustomizations.qll

```diff
@@ -106,6 +116,21 @@ module TaintedPath {
     }
   }
 
+  /**
+   * A replacement of the form `!strings.ReplaceAll(nd, "..")` or `!strings.ReplaceAll(nd, ".")`, considered as a sanitizer for
+   * path traversal.
+   */
+  class DotDotReplace extends Sanitizer {
+    DotDotReplace() {
+      exists(DataFlow::CallNode cleanCall, DataFlow::Node valueNode |
+        cleanCall = any(Function f | f.hasQualifiedName("strings", "ReplaceAll")).getACall() and
+        valueNode = cleanCall.getArgument(1) and
+        valueNode.asExpr().(StringLit).getValue() = ["..", "."] and
+        this = cleanCall.getResult()
+      )
+    }
+  }
```

Comment:
Use `StringOps::ReplaceAll` instead, like [here](https://github.com/github/codeql/blob/820c14577a9146175d792ee5111d2297050c50c1/go/ql/lib/semmle/go/security/LogInjectionCustomizations.qll#L36-L43). It covers more ways of doing the same thing.